### PR TITLE
Implement pre-auth check

### PR DESCRIPTION
**Pull request by yathindrakodithuwakku**
### Checklist
- [x] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/home.jsp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            var userAccessedPath = sessionStorage.getItem("userAccessedPath");
+            var isSilentSignInDisabled = userAccessedPath.includes("disable_silent_sign_in") || 
+                                            window.location.href.includes("disable_silent_sign_in");
+            var isInviteUserFlow = userAccessedPath.includes("invite_user");
+            var isApplicationsPath = userAccessedPath.includes("develop/applications") ||
+                                        window.location.href.includes("develop/applications") ||
+                                        sessionStorage.getItem("auth_callback_url_console").includes("develop/applications");
+            
+            if(isInviteUserFlow) {
+                window.location = window.location.origin;
```
Comment (brionmario) [inline apps/console/src/home.jsp:57]:
```suggestion
                window.location.href = window.location.origin;
```

File: apps/myaccount/src/index.jsp [hunk 2]
```diff
@@ -16,37 +16,86 @@
...
         </script>
-    </head>
-    <body>
-        <noscript>
-            You need to enable JavaScript to run this app.
-        </noscript>
-        <div id="root"></div>
     </body>
-</html>
+</html>
```
Comment (brionmario) [inline apps/myaccount/src/index.jsp:101]:
```suggestion
</html>

```

File: apps/console/src/index.jsp [hunk 3]
```diff
@@ -16,36 +16,101 @@
...
+                if(path) {
+                    return serverOrigin + path;
+                }
+
+                return serverOrigin;
+            }
+            
+            var auth = AsgardeoAuth.AsgardeoSPAClient.getInstance();
+
+            var authConfig = {
```
Comment (brionmario) [inline apps/console/src/index.jsp:57]: Let's see if we can get these values from deployment.config via webpack. This can be done in a followup PR.

**Review by brionmario — APPROVED** [review]

File: apps/console/src/home.jsp [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+* specific language governing permissions and limitations
+* under the License.
+-->
+
+<%= htmlWebpackPlugin.options.importUtil %>
+<%= htmlWebpackPlugin.options.importTenantPrefix %>
+<%= htmlWebpackPlugin.options.importSuperTenantConstant %>
+
+<jsp:scriptlet>
+    session.setAttribute("authCode",request.getParameter("code"));
```
Comment (thivi) [inline apps/console/src/home.jsp:24]:
```suggestion
    session.setAttribute("authCode", request.getParameter("code"));
```

File: apps/console/src/index.jsp [hunk 5]
```diff
@@ -16,36 +16,91 @@
...
+            var isSilentSignInDisabled = userAccessedPath.includes("disable_silent_sign_in");
+            var isTenantSwitchPath = userAccessedPath.includes("switch_tenant");
+            
+            if(isSilentSignInDisabled) {
+
+                if(isTenantSwitchPath) {
+                    auth.initialize(authConfig);
+                    auth.signIn();
+                } else {
+                    window.location.href = applicationDomain+'/authenticate?disable_silent_sign_in=true&invite_user=true';
```
Comment (thivi) [inline apps/console/src/index.jsp:89]:
```suggestion
                    window.location.href = applicationDomain + '/authenticate?disable_silent_sign_in=true&invite_user=true';
```

**Review by thivi — APPROVED** [review]

**Review by thivi — APPROVED** [review]

**Pull request merged**